### DISCUSSION TITLE
How to search relationship column

### DISCUSSION THREAD

**Discussion by aurawindsurfing** [filamentphp/filament#6079] [General]
Hi,

Looked everywhere but can not figure it out.

I have OrderResource that has such table:

```
   public static function table(Table $table): Table
    {
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('user_id')
                    ->searchable()
                    ->getStateUsing(fn($record) => $record->user->lastName.' '.$record->user->firstName)
```

Obviously I can only search by user id's

Is there any way to search by name of the related user? Like in forms with Select::

getSearchResultsUsing
or
getOptionLabelUsing

Thanks!

**Comment by aurawindsurfing** [comment]
@danharrin is that something that is possible?

**Comment by danharrin** [comment]
Name the column 'user.name' and then use `searchable(['first_name', 'last_name'])`

**Reply by aurawindsurfing** [reply]
Thank you!

**Comment by drcapins** [comment]
I have a situation a little bit complicated: the column is a MorphTo:
`
public function payable(): MorphTo
    {
        return $this->morphTo();
    }`

, so to show the value I need to write:
`
Tables\Columns\TextColumn::make('payable.user.full_name')
                    ->searchable(['lastname']),
`
but there's no way to search: if I try it comes:

Unknown column 'payable' in 'where clause' (Connection: mysql, SQL: select count(*) as aggregate from `payments` where `paid` < `cost` and (json_unquote(json_extract(`payable`, '$."user"."lastname"')) like %capor% or `payable_type` like %capor%))

It's too complicated?

**Reply by JairZijp** [reply]
Yeah same problem here.. Did you find a solution?

**Reply by drcapins** [reply]
nope

**Reply by JairZijp** [reply]
@drcapins this worked for me: https://www.answeroverflow.com/m/1128558694804766731

**Comment by jazzy1986** [comment]
what if i want to search like ->searchable(['id', 'client.name']), throw error when i add this

**Reply by jazzy1986** [reply]
found a way, using callback for searchable
```
  ->searchable(query: function (Builder $query, string $search): Builder {
                        return $query
                            ->where('id', 'like', "%{$search}%")
                            ->orWhereHas('client', function (Builder $clientQuery) use ($search) {
                                $clientQuery->where('name', 'like', "%{$search}%");
                            });
                    })
```
